main-app/app/scripts/providers/game-model.js

```diff
@@ -0,0 +1,126 @@
+(function () {
```

Comment:
In general the model is doing too much - something you should look at in later PRs.
This is just like OO work in C# - each "type" should have one responsbilty.
As a guide:
1. Abstract a provider out for handling player choice at the start
2. Any prior conversion of the API response should be a provider and - ideally - passed into the proxy call (mentioned below) rather than here
3. The game model should just handle game stuff - Initialising a new game, displaying a board, keeping track of whose turn it is etc...
4. For added polish - abstract default player types and noughts/crosses start positions into a constant provider - this actually makes the UI more testable as you can easily change it to try different starting configs.